ADR/pipeline-service/adr-0001.md

```diff
@@ -0,0 +1,46 @@
+# Service Account configuration for image builds in AppStudio 
+
+Date: Oct 5, 2022
+
+## Status
+
+Proposed
+
+## Context
+
+The `serviceAccount` used for running a `PipelineRun` workload in a workload cluster needs to have the right SCCs and linked credentials to be able to successfully execute a container image build. The linked credentials will mostly originate from the KCP workspace where the `PipelineRun` was synced from.
+Additionally, the SCC association needed for elevating capabilities required for image builds isn't something we'd prefer to expose the user to.
+
+
+
+## Decision
+
+* As a Pipeline Service platform feature, any service/user intending to build a container image would need to create/use the "image-builder" service account in the KCP workspace. 
+Use of any other service accounts will not lead to a successful assembling of the image.
+
+* The AppStudio Build controller would tap into this platform contract to create relevant PipelineRuns for building container images.
+
+## Consequences
+
+
+### Implementation - AppStudio Build Service Controller
+
+* The AppStudio Build controller creates an "image-builder" service account in the KCP workspace/namespace.
+* The AppStudio Build controller links the image pull secret ( previously created by SPI ) with the "image-builder" `serviceAccount` in the KCP workspace/namespace. Upon doing so, the 'updated' "image-builder" `serviceAccount` would get synced into the workload cluster.
+* The AppStudio Build controller" creates the PipelineRun referencing the "image-builder" service account as the one that the PipelineRun/TaskRun `Pods` would run as. Upong doing so, the PipelineRun would get synced into the Pipeline Service Workload cluster.
+* And that's it! Since the `[cluster]rolebinding` in the Workload cluster for all "image-builder" `serviceAccounts` would be pre-configured ( by the Pipeline Service ) with the SCC needed to build container images, the image should be successfully assembled. 
+* Additionally, since the service account had 
+the linked credentials, the container image should be successfully pushed as well.
+
+
+### Implementation - Pipeline Service 
+
+* The contract/API between users and the platform is the "image-builder" `serviceAccount` which consumers would be expected to use in the KCP workspace 
+for `PipelineRuns` if they wish to build container images.
+* The Pipeline Service platform would configure workload clusters with the `[cluster]rolebindings` necessary to have the "image-builder" `serviceAccount` 
+configured with a tailor-made SCC such that they are sufficiently capable of building container images. 
+
+
+### Security 
+
+* Users should not be able to alter role bindings in their `workspaces`/`namespaces` in a way that they are synced into the workload clusters. This will help prevent SCC escalations by the users/services whose workloads would be scheduled on the PipelineService workload cluster.
```

Comment:
Ideally Pipeline Service would use a ClusterRole, and ClusterRoleBindings SHOULD NOT be synced back from a workspace consuming any APIExport, as it would mean that a user's configuration could impact a service cluster.

Ideally a user should be free to add any Role/RoleBinding, secrets they want/need in their workspace, including to the SA account in order to satisfy the requirements for their workload.

Reply:
@s-urbaniak can workspace controllers create any `ClusterRole`? Can a user create any `ClusterRole` in their workspace?